e2e/cypress/integration/interactive_menu/basic_options.js

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/**
+* Note: This test requires webhook server running. Initiate `npm run start:webhook` to start.
+*/
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import users from '../../fixtures/users.json';
+import {getMessageMenusPayload} from '../../utils';
+
+const options = [
+    {text: 'Option 1', value: 'option1'},
+    {text: 'Option 2', value: 'option2'},
+    {text: 'Option 3', value: 'option3'},
+];
+const payload = getMessageMenusPayload({options});
+
+let channelId;
+let incomingWebhook;
+
+describe('MM-15887 Interactive menus - basic options', () => {
+    before(() => {
+        if (process.env.NODE_ENV !== 'qa') {
+            // Set AllowedUntrustedInternalConnections to localhost if running in development
+            const newSettings = {
+                ServiceSettings: {AllowedUntrustedInternalConnections: 'localhost'},
+            };
+            cy.apiUpdateConfig(newSettings);
+        }
+
+        // # Login as sysadmin and ensure that teammate name display setting us set to default 'username'
+        cy.apiLogin('sysadmin');
+        cy.apiSaveTeammateNameDisplayPreference('username');
+
+        // # Visit '/' and create incoming webhook
+        cy.visit('/');
+        cy.getCurrentChannelId().then((id) => {
+            channelId = id;
+
+            const newIncomingHook = {
+                channel_id: id,
+                channel_locked: true,
+                description: 'Incoming webhook interactive menu',
+                display_name: 'menuIn' + Date.now(),
+            };
+
+            cy.apiCreateWebhook(newIncomingHook).then((hook) => {
+                incomingWebhook = hook;
+            });
+        });
+    });
+
+    it('matches elements', () => {
+        // # Post an incoming webhook
+        cy.task('postIncomingWebhook', {url: incomingWebhook.url, data: payload});
+        cy.wait(TIMEOUTS.TINY);
```

Comment:
You might be able to omit this wait if you use an assertion off of the task. Maybe just `should('have.property', 'body')` could be sufficient enough? (It looks like there are a few other instances of this in this file as well).

It is just a tiny wait though, so I'll approve the PR and you can let me know if it's something that you feel is worth doing.

Reply:
Thanks Bob, I forgot about it and it should be it.  I'll update the PR.